src/test/rpc/Subscribe_test.cpp

```diff
@@ -651,6 +651,109 @@ class Subscribe_test : public beast::unit_test::suite
+        env.trust(USD(1000), alice, bob, carol);
+        env.close(NetClock::time_point{446000000s});
+
+        {
+            // add payments, but do no close until subscribed
+
+            // normal payments
+            env(pay(gw, alice, USD(50)));
+            ++numExpectedAvailable;
+            env(pay(gw, alice, XRP(50)));
+            ++numExpectedAvailable;
+
+            // partial payment
+            env(pay(gw, bob, USD(9999999)), txflags(tfPartialPayment));
+            ++numExpectedAvailable;
+            env.require(balance(bob, USD(1000)));
+
+            // failed payment
+            env(pay(gw, carol, USD(9999999)), ter(tecPATH_PARTIAL));
+            // do not incExpected - the payment failed
+            env.require(balance(carol, USD(0)));
+        }
+
+        auto wsc = makeWSClient(env.app().config());
+        Json::Value stream;
+
+        {
+            // RPC subscribe to ledger stream
+            stream[jss::streams] = Json::arrayValue;
+            stream[jss::streams].append("ledger");
+            stream[jss::accounts] = Json::arrayValue;
+            stream[jss::accounts].append(toBase58(alice.id()));
+            stream[jss::accounts].append(toBase58(bob.id()));
+            auto jv = wsc->invoke("subscribe", stream);
+            if (wsc->version() == 2)
+            {
+                BEAST_EXPECT(
+                    jv.isMember(jss::jsonrpc) && jv[jss::jsonrpc] == "2.0");
+                BEAST_EXPECT(
+                    jv.isMember(jss::ripplerpc) && jv[jss::ripplerpc] == "2.0");
+                BEAST_EXPECT(jv.isMember(jss::id) && jv[jss::id] == 5);
+            }
+            BEAST_EXPECT(jv[jss::result][jss::ledger_index] == 3);
+        }
+        {
+            env.close();
+            int numAvailable = 0;
+            // Check stream update
+            while (true)
+            {
+                auto const r = wsc->findMsg(1s, [&](auto const& jv) {
+                    return jv[jss::ledger_index] == 4;
+                });
+                if (!r)
+                    break;
+                if (!r->isMember("transaction"))
+                    continue;
+
+                auto const& t = (*r)["transaction"];
+                auto const& m = (*r)["meta"];
+                bool const expectDeliveredAmount =
+                    t["TransactionType"].asString() == "Payment" &&
+                    (*r)["engine_result"] == "tesSUCCESS";
+                bool const hasDeliveredAmount =
+                    m.isMember("delivered_amount") &&
+                    m["delivered_amount"] != "unavailable";
+                BEAST_EXPECT(expectDeliveredAmount == hasDeliveredAmount);
+                if (hasDeliveredAmount)
+                {
+                    ++numAvailable;
+                    if (m.isMember("DeliveredAmount"))
+                    {
+                        BEAST_EXPECT(
+                            m["delivered_amount"] == m["DeliveredAmount"]);
```

Comment:
Consider using `sfDeliveredAmount.jsonName`

Reply:
Er, I don't know about this one. For a test I think the string "DeliveredAmount" is fine `sfDeliveredAmount.jsonName` seems cumbersome. I left it as-is for now, but will change it if you feel strongly.

Reply:
The biggest advantage of `jsonName` is that the reference will be easier to find if we ever rename or restructure the field variable names.

Reply:
Fixed with: `Replace raw strings with sf<SomeName>.jsonName` commit